openedx/core/djangoapps/oauth_dispatch/docs/decisions/0008-use-asymmetric-jwts.rst

```diff
@@ -81,70 +68,56 @@ issuer, but with (the potential of) multiple signing keys stored in a JWT Set.
 .. _JSON Web Key Set (JWK Set): https://tools.ietf.org/html/draft-ietf-jose-json-web-key-36#section-5
 .. _site configuration: https://github.com/openedx/edx-platform/blob/af841336c7e39d634c238cd8a11c5a3a661aa9e2/openedx/core/djangoapps/site_configuration/__init__.py
 
-Example Code
-------------
+Features
+--------
 
 KeyPair Generation
 ~~~~~~~~~~~~~~~~~~
 
-Here is code for generating a keypair::
-
-    from Cryptodome.PublicKey import RSA
-    from jwkest import jwk
-
-    rsa_key = RSA.generate(2048)
-    rsa_jwk = jwk.RSAKey(kid="your_key_id", key=rsa_key)
-
-To serialize the **public key** in a `JSON Web Key Set (JWK Set)`_::
-
-    public_keys = jwk.KEYS()
-    public_keys.append(rsa_jwk)
-    serialized_public_keys_json = public_keys.dump_jwks()
-
-and its sample output::
-
-    {
-        "keys": [
-            {
-                "kid": "your_key_id",
-                "e": "strawberry",
-                "kty": "RSA",
-                "n": "something"
-            }
-        ]
-    }
-
-To serialize the **keypair** as a JWK::
-
-    serialized_keypair = rsa_jwk.serialize(private=True)
-    serialized_keypair_json = json.dumps(serialized_keypair)
-
-and its sample output::
-
-    {
-        "e": "strawberry",
-        "d": "apple",
-        "n": "banana",
-        "q": "pear",
-        "p": "plum",
-        "kid": "your_key_id",
-        "kty": "RSA"
-    }
+Please have a look at ``openedx/core/djangoapps/oauth_dispatch/management/commands/generate_jwt_signing_key.py``
+to get better understanding how to generate keypair using ``PyJWT``.
+
+The public and private keypair would be similar to the following::
+
+    ## Public keyset
+    """
+        {
+            "keys": [
+                {
+                    "kty": "RSA",
+                    "key_ops": ["verify"],
+                    "n": "banana",
+                    "e": "AQAB",
+                    "kid": "your_key_id"
+                }
+            ]
+        }
+    """
+
+
+    ## Private key
+    """
+        {
+            "kty": "RSA",
+            "key_ops": ["sign"],
+            "n": "banana",
+            "e": "AQAB",
+            "d": "apple",
+            "p": "peach",
+            "q": "pear",
+            "dp": "palm",
+            "dq": "pineapple",
+            "qi": "watermelon",
+            "kid": "your_key_id"
```

Comment:
I think this would also be a good opportunity to switch away from the placeholder words to something that is more visually obvious as a placeholder:

```suggestion
                    "kty": "RSA",
                    "key_ops": ["verify"],
                    "n": "...",
                    "e": "...",
                    "kid": "your_key_id"
                }
            ]
        }
    """


    ## Private key
    """
        {
            "kty": "RSA",
            "key_ops": ["sign"],
            "n": "...",
            "e": "...",
            "d": "...",
            "p": "...",
            "q": "...",
            "dp": "...",
            "dq": "...",
            "qi": "...",
            "kid": "your_key_id"
```

Reply:
We should also replace all these tab characters with consistent space-based indentation.